src/store/product/reducer.js

```diff
@@ -1,19 +1,37 @@
 import actions from './actionTypes';
 
-const initialState = { productAdded: true };
+let products = JSON.parse(localStorage.getItem('products'));
+const initialState = {
+  productAdded: true,
```

Comment:
you can combine productAdded and productsReceived into one boolean value called loading